include/swift/Reflection/ReflectionContext.h

```diff
@@ -801,7 +801,50 @@ class ReflectionContext
     }
 
     case RecordKind::MultiPayloadEnum: {
-      // TODO: Support multipayload enums
+      // Collect basic statistics about the enum
+      unsigned long PayloadCaseCount = 0;
+      unsigned long NonPayloadCaseCount = 0;
+      unsigned long PayloadSize = 0;
+      for (auto Field : Fields) {
+        if (Field.TR != 0) {
+          PayloadCaseCount += 1;
+          if (Field.TI.getSize() > PayloadSize) {
+            PayloadSize = Field.TI.getSize();
+          }
+        } else {
+          NonPayloadCaseCount += 1;
+        }
+      }
+      if (EnumSize > PayloadSize) {
```

Comment:
@slavapestov   If `EnumSize > PayloadSize`, does that ensure that the multi-payload enum is using a separate tag and that tag starts immediately following `PayloadSize`?   If so, then I think this PR can be standalone, as it will properly reject multi-payload enums that it doesn't understand.

Support for enums that use spare bits can be added as a separate PR.

Reply:
Again we need to plumb this through from lowering (as we already discussed in chat)

Reply:
Good point.  Unless you object, I'll make that change in a follow-up PR.